src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1438,14 +1438,29 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
         }
     }
 
-    private function registerMessengerConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    private function registerMessengerConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader, array $serializerConfig)
     {
         if (!interface_exists(MessageBusInterface::class)) {
             throw new LogicException('Messenger support cannot be enabled as the Messenger component is not installed.');
         }
 
         $loader->load('messenger.xml');
 
+        if ($this->isConfigEnabled($container, $config['transport']['serializer'])) {
+            if (count($config['adapters']) > 0 && !$this->isConfigEnabled($container, $serializerConfig)) {
```

Comment:
Is the `count($config['adapters']) > 0` required? 
To me, as soon as the `$config['transport']['serializer']` is enabled without serializer support enabled we should throw anyway. We don't really care if it's used by an adapter or not at this point:
in any case, `messenger.transport.serializer` will never be automatically enabled, unless using flex and both Serializer and Messenger component are installed (so everything is fine). 
If enabled without without the Serializer installed/enabled, it means the developer explicitly enabled `messenger.transport.serializer` or disabled `serializer`. 😃

Reply:
I was actually changing this (see the last commit). Now this makes sense (by have the Symfony Serialized based Transport enabled no matter what). It's important to have the `count($config['adapters'])` so that it doesn't throw anything if you don't use any adapter (i.e. don't require an encoder/decoder).